### DISCUSSION TITLE
Help with release triggers

### DISCUSSION THREAD

**Discussion by Eddi-S** [community/community#144221] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a github action with the following triggers
```
on:
  release:
    types: [created, published, released] 
```
But the action is never triggered when I create a release through the ui:
![image](https://github.com/user-attachments/assets/43ba49b3-acfe-4494-baeb-0c570fbe3663)
What am I missing?

**Comment by gr1shan1a** [comment]
Hi, there!

At first, ensure that your workflow file is correctly named and placed in the .github/workflows directory of your repository.

check if GitHub Actions are enabled for your repository and ensure that you have necessary permissions for this

There are also **Logs**, so check them and keep up to date

**Comment by gr1shan1a** [comment]
also there might be some conflicts with GitHub action - check version and compatibility

**Comment by M-tech-cmd** [comment]
It sounds like you’re having trouble with your GitHub Actions workflow not triggering on release events.

Ensure your workflow file is located in the .github/workflows/ directory of your repository](https://github.com/orgs/community/discussions/54574)[1](https://github.com/orgs/community/discussions/54574).

Correct Event Types: Verify that you are using the correct event types. [For releases, you should use created, published, or released as you have done]

(https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow)[2](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow).

[GitHub Actions Enabled: Make sure GitHub Actions is enabled for your repository](https://github.com/orgs/community/discussions/54574)[1](https://github.com/orgs/community/discussions/54574).

Token Permissions: If you are using the default GITHUB_TOKEN, it might not trigger workflows for certain events. [Try using a personal access token (PAT) instead](https://github.com/orgs/community/discussions/25281)[3](https://github.com/orgs/community/discussions/25281)[2](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow).

[Check Logs: Look at the “Actions” tab in your repository to see if there are any logs or errors that can give you more information](https://github.com/orgs/community/discussions/54574)[1](https://github.com/orgs/community/discussions/54574).

[Branch: Ensure that the workflow file is present on the default branch of your repository](https://devopsjournal.io/blog/2022/08/12/workflows-not-starting)[4](https://devopsjournal.io/blog/2022/08/12/workflows-not-starting).

**Comment by Eddi-S** [comment, marked as answer]
I finally figured out what was the problem since I was creating release for existing tags which does not trigger the events.
Info from copilot:
Creating a new tag during the release process triggers the published event, which in turn triggers the workflow. When you use an existing tag, GitHub considers the release as published at the time the tag was initially created. Therefore, it doesn't trigger the published event again, and the workflow doesn't run.

So, if you want your workflow to run on release creation, it's a good practice to create a new tag during the release process. This ensures that the published event is triggered and your workflow runs as expected.

**Reply by M-tech-cmd** [reply]
about the  triggers here are the additional things you might want to check:

Event Types: Ensure that the release events you’re creating match the types specified (created, published, released). Sometimes, the event might not be exactly what you expect.

Repository Settings: Double-check your repository settings to ensure that GitHub Actions are enabled and there are no restrictions on the types of actions that can run.

Permissions: If your release is being created by another action or a bot, make sure it has the necessary permissions. Using a personal access token (PAT) with the right scopes can help.

YAML Syntax: YAML can be picky about indentation and formatting. Make sure there are no hidden syntax errors in your workflow file.

Logs: Look at the logs in the “Actions” tab of your repository. They can provide valuable clues about why the workflow isn’t triggering

an example on how your workflow file might look:

name: Release Workflow

on:
  release:
    types: [created, published, released]

jobs:
  release:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      # Add your steps here
